src/TTextEdit.cpp

```diff
@@ -391,20 +392,26 @@ void TTextEdit::drawLine(QPainter& painter, int lineNumber, int lineOfScreen) co
         TChar timeStampStyle(QColor(200, 150, 0), QColor(22, 22, 22));
         QString timestamp(mpBuffer->timeBuffer.at(lineNumber));
         for (const QChar c : timestamp) {
-            cursor.setX(cursor.x() + drawGrapheme(painter, cursor, c, 0, timeStampStyle));
+            cursor.setX(cursor.x() + drawGrapheme(painter, cursor, c, 0, timeStampStyle).first);
         }
     }
 
     int columnWithOutTimestamp = 0;
+    int spaceForWidestGlyph = 0;
     for (int indexOfChar = 0, total = lineText.size(); indexOfChar < total;) {
         int nextBoundary = boundaryFinder.toNextBoundary();
 
         TChar& charStyle = mpBuffer->buffer.at(lineNumber).at(indexOfChar);
-        int graphemeWidth = drawGrapheme(painter, cursor, lineText.mid(indexOfChar, nextBoundary - indexOfChar), columnWithOutTimestamp, charStyle);
-        cursor.setX(cursor.x() + graphemeWidth);
+        QString grapheme{lineText.mid(indexOfChar, nextBoundary - indexOfChar)};
+        QPair<int, int> graphemeWidths = drawGrapheme(painter, cursor, grapheme, columnWithOutTimestamp, charStyle);
+        cursor.setX(cursor.x() + graphemeWidths.first);
```

Comment:
`QPair<int, int>` and `.first` give no semantic information about what this thing is!

Use [structured binding](https://en.cppreference.com/w/cpp/language/structured_binding) to help with this. It's the `[]` we use in other places already.